constraints/miden-vm/hash.air

```diff
@@ -260,9 +286,37 @@ ev hash_sibling_table_constraints(main: [s[3], r, h[12], i], aux: [p1]):
     let v_sibling = $alpha[0] + $alpha[3] * i + b * v_b + (1 - b) * v_c
 
     # Enforce constraint for updating p1.
-    enf p1' * ((get_f_mv(s) + get_f_mva(s)) * v_sibling + 1 - (get_f_mv(s) + get_f_mva(s))) = 
-        p1 * ((get_f_mu(s) + get_f_mua(s)) * v_sibling + 1 - (get_f_mu(s) + get_f_mua(s)))
+    # Constraint degree: 7
+    enf p1' * ((f_mv + f_mva) * v_sibling + 1 - (f_mv + f_mva)) = 
+        p1 * ((f_mu + f_mua) * v_sibling + 1 - (f_mu + f_mua))
 
     # Enforce that computation of the old Merkle root is immediately followed by the computation of 
     # the new Merkle root.
-    enf (get_f_bp(s) + get_f_mp(s) + get_f_mv(s) * (1 - p1)) = 0
\ No newline at end of file
+    # Constraint degree: 5
+    enf (f_bp + f_mp + f_mv) * (1 - p1) = 0
+
+
+### Hash Chiplet Air Constraints ##################################################################
+
+# Enforces the constraints on the hash chiplet, given the columns of the hash execution trace.
+#
+# Max constraint degree: 7
+ev hash_chiplet([s[3], r, h[12], i]):
+    ## Row address constraint ##
+    # TODO: Apply row address constraints:
+    # 1. Boundary constraint `enf r.first = 1`
+    # 2. Transition constraint. It requires chiplets module's selector flag s0.
```

Comment:
I think we can add this transition constraint, I'm just not sure where we should place `chip_s0` chiplet module's selector flag -- to the first (`main`) segment, or to the second (`aux`).